persist/fs/write.go

```diff
@@ -117,33 +129,41 @@ func NewWriter(opts Options) (FileSetWriter, error) {
 // Open initializes the internal state for writing to the given shard,
 // specifically creating the shard directory if it doesn't exist, and
 // opening / truncating files associated with that shard for writing.
-func (w *writer) Open(
-	namespace ident.ID,
-	blockSize time.Duration,
-	shard uint32,
-	blockStart time.Time,
-) error {
-	shardDir := ShardDirPath(w.filePathPrefix, namespace, shard)
-	if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
-		return err
+func (w *writer) Open(opts WriterOpenOptions) error {
+	var shardDir string
+	var fileTimestampUnixNano time.Time
+	if opts.IsSnapshot {
+		shardDir = ShardSnapshotsDirPath(w.filePathPrefix, opts.Namespace, opts.Shard)
+		if err := os.MkdirAll(shardDir, w.newDirectoryMode); err != nil {
```

Comment:
you can capture the time you want to save then return weather there was an error for making all the directories, as the exact same chunk is repeated on 143

Reply:
yeah agreed dont need to duplciate the mkdir all